modules/rostests/kmtests/ntos_ex/ExHardError.c

```diff
@@ -116,12 +117,12 @@ TestHardError(
     CheckHardError(0x40000013,                  0, OptionYesNoCancel,       STATUS_SUCCESS,            ResponseNo,             0, 0);                          // outputs a box :|
     CheckHardError(0x40000013,                  0, OptionYesNoCancel,       STATUS_SUCCESS,            ResponseCancel,         0, 0);                          // outputs a box :|
     }
-    CheckHardError(0x40000009,                  0, 9,                       STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000a,                  0, 10,                      STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000b,                  0, 11,                      STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000c,                  0, 12,                      STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000d,                  0, MAXULONG / 2 + 1,        STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
-    CheckHardError(0x4000000d,                  0, MAXULONG,                STATUS_SUCCESS,            ResponseNotHandled,     0, 0);
+    CheckHardError(0x40000009,                  0, 9,                       STATUS_SUCCESS,            ResponseReturnToCaller, 0, 0);
```

Comment:
I have no idea what's going on with these. My Win2003 returns `ResponseNotHandled` for these, and the logic of the code in csrsrv/winsrv makes it pretty clear that that's what we _should_ be getting here. I need to do some analysis of what's going on on the WHS bot to understand why it's returning `ResponseReturnToCaller`.

Reply:
Could it be related to the non-initialized variable vulnerability in NtRaiseHardError (on Windows) that you mentioned on #kernel-dev on Mattermost?

Reply:
That one is triggered by the `OptionShutdownSystem` call above. These here should successfully make it all the way to winsrv and be failed there with `ResponseNotHandled`.